transmogrifier/sources/transformer.py

```diff
@@ -74,17 +63,22 @@ def __init__(
         self.processed_record_count: int = 0
         self.transformed_record_count: int = 0
         self.skipped_record_count: int = 0
+        self.error_record_count: int = 0
```

Comment:
How is are `skipped_record_count` and `error_record_count` related? 🤔 If I remember correctly, a record is **skipped** if the source record is lacking sufficient data for a transform.

Reply:
`error_record_count` is when an **unhandled** exception is thrown during transformation.  Currently, it bubbles all the way up and halts Transmogrifier.

Now, we'll swallow that exception and the actual `action` column in the dataset will read `error`, so we can pinpoint records that have had errors (and therefore didn't get indexed). 

I'd imagine that we might eventually set a threshold, so if Transmog encounters 100 errors in a row -- or something -- we halt the program.  But in 3 million records from Alma, if _one_ record has a weird bug, it'd be nice to complete the job but we'll surface this in the dataset.  

Or heck... maybe even better, we write a Sentry message at the end of Transmog if _any_ errors, but it still completed successfully and wrote to the dataset.